Refactor sl-mpc-mate

left-arm opened the pull request:
- feat-ot-variant
- Replace message::InvalidMessage with coord::MessageSendError
- Remove message::NonceCounter
- Remove reexport of ed25519_dalek x25519_dalek types
- Remove unsafe code
- Remove unused deps
- Generalized `factorial_range()`. It works for any PrimeField now
  instead of U256 as before.
- Make table of recomputed factorials in range 0..20
- Factor out `polynomial_coeff_multipliers_iter()` from
  `polynomial_coeff_multipliers()` and rewrote it to avoid
  writing to result vector twice
- add tests

left-arm committed:
Refactor sl-mpc-mate

- Replace message::InvalidMessage with coord::MessageSendError
- Remove message::NonceCounter
- Remove reexport of ed25519_dalek x25519_dalek types
- Remove unused deps

left-arm committed:
Make rayon optional

left-arm committed:
Make MessageRelayService::connect() async

left-arm committed:
Add method coord::Relay::ask(msg_id, ttl)

- remove blanket implementation of Relay
- use it BufferedMsgRelay
- small refactoring

leontiad reviewed (approved):
I would suggest removing everything under coord and place it under the networking crate in dkls23 repo since these are not related with cryptography code under sl-crypto

left-arm committed:
Remove 'static lifetime bound for Relay trait

And give method new() to MaybeFeed

left-arm committed:
Update base64 to 0.22

left-arm committed:
derive Zeroize for EndemicOTReceiver

left-arm committed:
Fix issue #9

left-arm committed:
Refactor MsgHdr and related structures

- MsgHdr is array of MESSAGE_HEADER_SIZE bytes
- provide accessors for MsgHdr::id, MsgHdr::ttl
- new field MsgHdr::flags
- new method MsgHdr::encode()
- implement TryFrom<&'a [u8]> for &'a MsgHdr
- adjust rest of code to new MsgHdr API.
- add a bit of documentation

eclipse7 committed:
[+] rvole_ot_variant

left-arm committed:
Remove unsafe code

left-arm committed:
Remove empty or broken doc comments

left-arm committed:
Remove unnecessary lifetime parameter.

left-arm committed:
Refactoring sl_mpc_mate::math

- Generalized `factorial_range()`. It works for any PrimeField now
  instead of U256 as before.
- Make table of recomputed factorials in range 0..20
- Factor out `polynomial_coeff_multipliers_iter()` from
  `polynomial_coeff_multipliers()` and rewrote it to avoid
  writing to result vector twice
- add tests

leontiad reviewed (commented):
Can we move coord.rs and coord/* out of here? It has nothing to do with math or crypto
